### ISSUE TITLE
Better PropType Checking

### ISSUE THREAD

**Issue by jehoshua02** [deseretdigital-ui/ksl-header#52]
So ... I learned about this `React.PropType.shape({ ... })`. I should have known about it since I read the React docs, but for some reason my brain never reminded me to use it.

**Comment by janiv** [comment]
:+1: Duh, how did I overlook this in the docs too?

**Comment by jehoshua02** [comment]
Yeah, in [another project](http://jsbin.com/buwibe/11/edit?js,output) I wanted to keep the data separate from other props (See `Habit` component) so the `React.PropTypes.shape` came in quite handy.
